src/test/java/study/SetTest.java

```diff
@@ -0,0 +1,57 @@
+package study;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class SetTest {
+
+    private Set<Integer> numbers;
+
+    @BeforeEach
+    public void setUp() {
+        numbers = new HashSet<>();
+        numbers.add(1);
+        numbers.add(1);
+        numbers.add(2);
+        numbers.add(3);
+    }
+
+    @Test
+    @DisplayName("HashSet 의 크기 테스트")
+    public void sizeTest() {
+        assertThat(numbers.size()).isEqualTo(3);
+    }
+
+    @Test
+    @DisplayName("HashSet 이 갖고 있는 값 테스트")
+    public void containsTest() {
+        assertThat(numbers.contains(1)).isTrue();
+        assertThat(numbers.contains(2)).isTrue();
+        assertThat(numbers.contains(3)).isTrue();
+    }
+
+    @ParameterizedTest
+    @DisplayName("중복 코드를 제거한 HashSet 이 갖고 있는 값 테스트")
+    @ValueSource(ints = {1, 2, 3})
+    public void containsParameterizedTest(final int number) {
+        assertTrue(numbers.contains(number));
+    }
+
+    @ParameterizedTest
+    @DisplayName("CsvSource를 이용한 입력 값에 따른 결과 값 테스트")
+    @CsvSource(value = {"1,true", "2,true", "3,true", "4,false", "5,false"})
+    public void size1Test(final int number, final String bool) {
```

Comment:
```suggestion
    public void size1Test(final int number, final boolean result) {
```
1. CsvSource 에서 자동으로 boolean 으로 판별이 가능하므로 String 타입으로 받은 후 변환 과정은 없어져도 될 거 같아요.
2. 파라미터의 네이밍은 타입명보다는 좀 더 의미 있는 이름으로 지어보는 것은 어떨까요~?

Reply:
CsvSource는 당연히 String, String 으로 받을 줄 알았는데 의외네요. 
생각지도 못했습니다!